### PR TITLE
Improve description of conversions

### LINKED ISSUE #13811: Improve the "Conversion rules" documentation
The documentation lacks understandable definitions of beta, delta, iota and zeta reduction in language that will be understood by Coq users who are not intimately familiar with CIC theory.  These are essential concepts.  They need to be explained precisely and clearly using Coq syntax rather than mathematical notation (or perhaps in addition to mathematical notation) with some concrete examples that use Coq's reduction tactics.

The significance of eta-expansion is unclear.  If it's an important concept that users need to understand, it should be explained clearly, too.  The text mentions βδιζη-convertible, including eta.

Not sure what to make of the "Convertibility" section.  If it can be simplified and made more understandable to non-CIC experts that would be good.

Attn: @JasonGross

### PR DESCRIPTION
**Pull request by jfehrle**
Fixes #13811.

I added some introductory text but I'm not sure what to do with the existing material.  I could leave that as-is.  I think it would take a lot of time for me to understand the details well enough to change much of the existing material.

Attn: @JasonGross

### CODE REVIEW

File: doc/sphinx/language/core/conversion.rst [hunk 1]
```diff
@@ -3,11 +3,75 @@
...
+:n:`@ident` with :n:`@term__1` wherever :n:`@ident` appears in :n:`@term__2`.
+The resulting term may be longer or shorter than the original.
+
+.. coqtop:: all
+
+   Eval cbv zeta in let i := 1 in i + i.
+
+Expansions are reductions applied in the opposite direction,
+for example expanding `2 + 2` to `let i := 2 in i + i`.  While applying
+reductions (except for eta-reduction) gives a unique result , the associated
```
Comment (JasonGross) [inline doc/sphinx/language/core/conversion.rst:26]: Eta reduction gives a unique result, but the result is not necessarily of a type that's convertible with the type of the expanded form.

File: doc/sphinx/language/core/conversion.rst [hunk 2]
```diff
@@ -3,11 +3,75 @@
...
+expansion may not give a unique result.  For example, `2 + 2` could also be
+expanded to `let i := 2 in i + 2`.  Expansions that give a unique
+result are also referred to as *contractions*.
```
Comment (JasonGross) [inline doc/sphinx/language/core/conversion.rst:29]: This isn't right; contraction is the inverse of expansion.  Reductions which are both guaranteed to always make the term smaller and which have a unique inverse expansion are called contractions.  (I'm not sure if both these conditions are necessary, though.). Perhaps it would be safer to call out eta contraction by name?

File: doc/sphinx/language/core/conversion.rst [hunk 3]
```diff
@@ -3,11 +3,75 @@
...
+        - Description
+
+      * - beta-reduction
+        - eliminates `fun`
+
+      * - delta-reduction
+        - replaces a defined symbol with its definition
+
+      * - cofix-reduction
+        - replace a `cofix` symbol with its definition
```
Comment (JasonGross) [inline doc/sphinx/language/core/conversion.rst:54]: Not quite.  When a `cofix` is applied to arguments such that it's structural argument is headed by a constructor, the cofix reduction will replace the cofix with a fun (leaving a beta redex) and replace any recursive calls with the original `cofix` term

File: doc/sphinx/language/core/conversion.rst [hunk 4]
```diff
@@ -3,11 +3,75 @@
...
+
+      * - fix-reduction
+        - replaces a `fix` symbol with its definition
```
Comment (JasonGross) [inline doc/sphinx/language/core/conversion.rst:57]: Not quite.  When a `fix` is applied to arguments such that it's structural argument is headed by a constructor, the fix reduction will replace the fix with a fun (leaving a beta redex) and replace any recursive calls with the original `fix` term

File: doc/sphinx/language/core/conversion.rst (hunk 3 above)
Comment (JasonGross) [inline doc/sphinx/language/core/conversion.rst:54]: Not quite.  When a `cofix` is the discriminee of a match, the cofix reduction will replace the cofix with it's body (possibly leaving a beta redex, but often just leaving a constructor) and replace any recursive calls with the original `cofix` term

File: doc/sphinx/language/core/conversion.rst [hunk 5]
```diff
@@ -3,11 +3,75 @@
...
+        - eliminates `match`
+
+      * - iota-reduction
+        - match-, fix- and cofix-reduction together
+
+      * - zeta-reduction
+        - eliminates `let`
+
+      * - eta-expansion
+        - TBA
```
Comment (JasonGross) [inline doc/sphinx/language/core/conversion.rst:69]: Replaces a term `f` of type `forall a : A, B` with `fun x : A => f x`

Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:69]: Hard to create a convincing example in coqtop.  Could I have something vaguely like `forall a : A, B = fun x : A => f` which `cbv` fails to convert (lacking eta) while, say, `reflexivity` includes eta and so show they are equivalent?

Comment (JasonGross) [reply doc/sphinx/language/core/conversion.rst:69]: The canonical example is the one I gave in the issue tracker involving composition of a variable with the identity function.  I think that should be the example that gets included.

Note also that there is both function eta and record eta, and the latter is controlled with Set Primitive Projections.  Record eta contraction is turning (fst x, snd x) into x and similar for other records.

File: doc/sphinx/language/core/conversion.rst (hunk 2 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:29]: Fixed the wording.

From your comment above, I thought eta-reduction was not useful in Coq.  Also IIUC not used to determine convertibility nor available in `cbv` et al.  If so, not worth mentioning.

For this chapter, I will be satisfied if I can give a reasonable concrete description of how the conversion rules rewrite terms without contradicting the theory behind them.  Hopefully explaining a bit about why they're useful, too.  Explaining how the theory works, no, not at this time.

File: doc/sphinx/language/core/conversion.rst (hunk 4 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:57]: It does remove the `fix` from the term, right?  If so, I could just say "replaces a fix symbol" here and give an example or two of the transformation later in the chapter.

Is this an accurate description of a beta-redex https://en.wikipedia.org/wiki/Beta_normal_form?  `(λx. A)M` is `(fun x => A) M` in Gallina?

File: doc/sphinx/language/core/conversion.rst (hunk 3 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:54]: Basically same comments as for `fix` above.

File: doc/sphinx/language/core/conversion.rst (hunk 5 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:69]: BTW, I haven't studied the typing rules much yet, so I don't know how to construct a term that has the type `forall a : A, B`.

File: doc/sphinx/language/core/conversion.rst (hunk 4 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:57]: I assume there's no relationship between `beta-redex` and `beta-reduction`.

File: doc/sphinx/language/core/conversion.rst (hunk 5 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:69]: > The canonical example is the one I gave in the issue tracker involving composition of a variable with the identity function ...

Which PR?  I think not #2860, whichi is eta-reduction rather than eta-expansion.

Comment (JasonGross) [reply doc/sphinx/language/core/conversion.rst:69]: > Which PR? I think not #2860, which is eta-reduction rather than eta-expansion.

In https://github.com/coq/coq/issues/13811#issuecomment-771104374.  Note that while "reduction" is directed (towards normalization), supporting a reduction rule in conversion necessarily means supporting both directions, because "convertible" is an equivalence relation.

The reason eta-expansion is useful is because putting everything in eta-long form (fully eta-expanded) does not break typing and allows untyped conversion to support eta.  eta-contraction, on the other hand, is much harder to implemented when you're not carrying around types.  It's primarily useful for displaying things in a nicer format (I imagine that `simpl` would want to do eta-contraction, while `cbv` would not), and occasionally for de-duplicating work in a non-shared reduction strategy (computing `(fst x, snd x)` where `x` is a complicated term may compute `x` twice rather than once).

Comment (JasonGross) [reply doc/sphinx/language/core/conversion.rst:69]: > BTW, I haven't studied the typing rules much yet, so I don't know how to construct a term that has the type `forall a : A, B`.

You can access to one by writing `fun f : (forall a : A, B) => ...`.  Or via `Axiom`.  Alternatively, if you want a concrete term for some particular choice of `A` and `B`, `Nat.add` is one, and `id` (or `fun a => a`) is another.  Really I'm just talking about any term which is a function i.e., which can take arguments / be applied to things.

File: doc/sphinx/language/core/conversion.rst (hunk 4 above)
Comment (JasonGross) [reply doc/sphinx/language/core/conversion.rst:57]: > It does remove the `fix` from the term, right?

No, it sort-of pushes the `fix` inwards.  If you have `fix f := F f`, then doing `fix`-reduction corresponds to replacing this with `F (fix f := F f)`.

> Is this an accurate description of a beta-redex https://en.wikipedia.org/wiki/Beta_normal_form? `(λx. A)M` is `(fun x => A) M` in Gallina?

Yes to both.

Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:57]: Look at what I wrote for this and `cofix`.  This is meant to a terse overview, not to give full details.

File: doc/sphinx/language/core/conversion.rst (hunk 5 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:69]: AFAIK there's no tactic that generates and displays the eta-expansion of a term, so not sure it makes sense for me to try give an example--hard for me to know if it's correct.

**Comment by jfehrle** [comment]
Updated.  I think this is sufficient; I don't want to try to explain too rigorously.

File: doc/sphinx/language/core/conversion.rst [hunk 6]
```diff
@@ -4,11 +4,83 @@ Conversion rules
...
+   Eval cbv zeta in let i := 1 in i + i.
+
+Expansions are reductions applied in the opposite direction,
+for example expanding `2 + 2` to `let i := 2 in i + i`.  While applying
+reductions gives a unique result, the associated
+expansion may not be unique.  For example, `2 + 2` could also be
+expanded to `let i := 2 in i + 2`.  Reductions that have a unique inverse
+expansion are also referred to as *contractions*.
+
+The normal form is defined as the result of applying a particular
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:37]: Same question for normal form.

File: doc/sphinx/language/core/conversion.rst [hunk 7]
```diff
@@ -4,11 +4,83 @@ Conversion rules
...
+set of conversion rules (beta-, delta-, iota- and zeta-reduction and eta-expansion)
+repeatedly until it's no longer possible to apply any of them.
+
+Some reductions can be considered computations, for example reducing `2*3+4` with
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:41]: What do you mean by this? Isn't applying reduction rules always a form of computation?

File: doc/sphinx/language/core/conversion.rst [hunk 8]
```diff
@@ -4,11 +4,83 @@ Conversion rules
...
+`cbv beta delta iota` to `10`, which requires applying several reduction rules
+repeatedly.
+
+The useful conversion rules are:
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:45]: "useful" or "available"?

File: doc/sphinx/language/core/conversion.rst [hunk 9]
```diff
@@ -4,11 +4,83 @@ Conversion rules
...
+
+      * - beta-reduction
+        - eliminates `fun`
+
+      * - delta-reduction
+        - replaces a defined symbol with its definition
+
+      * - cofix-reduction
+        - replaces a `cofix` symbol with a :term:`beta-redex`;
+          recursive calls to the symbol are replaced with the `cofix` term
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:61]: Is there a way to split this on two lines in the table? Right now, you have to scroll right to read the whole thing.

File: doc/sphinx/language/core/conversion.rst [hunk 10]
```diff
@@ -4,11 +4,83 @@ Conversion rules
...
+Conversion rules can be used
+to determine if two terms are convertible by
+converting both terms to a normal form, then verifying they are syntactically
+equal (ignoring differences in the names of bound variables by
+:term:`alpha-convertibility <alpha-convertible>`).
+
 See :ref:`applyingconversionrules`,
 which describes tactics that apply these conversion rules.
 
+Conversion rules can be reductions or expansions.
```
[the remark this answers is not in the document]
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:19]: Got it.  There were similar changes in #13707 that I didn't clean up after the rebase.

File: doc/sphinx/language/core/conversion.rst [hunk 11]
```diff
@@ -4,11 +4,83 @@ Conversion rules
...
+.. coqtop:: all
+
+   Eval cbv zeta in let i := 1 in i + i.
+
+Expansions are reductions applied in the opposite direction,
+for example expanding `2 + 2` to `let i := 2 in i + i`.  While applying
+reductions gives a unique result, the associated
+expansion may not be unique.  For example, `2 + 2` could also be
+expanded to `let i := 2 in i + 2`.  Reductions that have a unique inverse
+expansion are also referred to as *contractions*.
```
[the remark this answers is not in the document]
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:35]: I added these as defined terms.  I can do a bulk change to add hyperlinks in the text, but I think I would lean on you again to identify references that shouldn't be hyperlinked.  I see 41 uses of "expansion", 7 of "contraction" and 146 of "reduction".

File: doc/sphinx/language/core/conversion.rst (hunk 7 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:41]: Perhaps I should have said that a series of reductions may correspond to evaluating an expression (as in a programming language) where you end up with a constant rather a general term.  Certainly some computations can run for a very long time.  What do you suggest?

File: doc/sphinx/language/core/conversion.rst (hunk 8 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:45]: The table describes conversions in general rather than those just those that Coq supports.  IIUC most of the expansions are not unique, so they're not useful, nor are they  available in Coq tactics.  Also there's no tactic for doing eta-expansion, unlike the reductions.

By "useful" I was thinking unique/invertible.  Maybe I should say that instead.  I could add a clause saying they can all be applied with conversion tactics, except for eta-expansion.

**Comment by jfehrle** [comment]
Updated.

File: doc/sphinx/language/core/conversion.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: Maybe that should be done incrementally as we go over the text instead.

File: doc/sphinx/language/core/conversion.rst (hunk 6 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst:34]: Indeed. Head normal form or weak-head normal forms are just specialized versions of a normal form where we don't reduce everywhere.

File: doc/sphinx/language/core/conversion.rst (hunk 7 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst:38]: Maybe just clarify that sometimes the normal forms obtained are values and sometimes computations get blocked by variables / axioms / opaque constants, contrary to usual programming languages.

File: doc/sphinx/language/core/conversion.rst (hunk 6 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:34]: So I won't add it as a defined term.  Right?

File: doc/sphinx/language/core/conversion.rst (hunk 7 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:38]: "Sometimes the result of a reduction tactic will be a simple value, for example reducing `2*3+4` with `cbv beta delta iota` to `10`, which requires applying several
reduction rules repeatedly.  In other cases, it may yield an expression containing variables, axioms or opaque contants that can't be reduced."

**Comment by jfehrle** [comment]
Updated.

File: doc/sphinx/language/core/conversion.rst [hunk 12]
```diff
@@ -142,6 +212,29 @@ for :math:`x` an arbitrary variable name fresh in :math:`t`.
    because the type of the reduced term :math:`∀ x:\Type(2),~\Type(1)` would not be
    convertible to the type of the original term :math:`∀ x:\Type(1),~\Type(1)`.
 
+Examples
+~~~~~~~~
+
+   .. example:: Simple delta, fix, beta and match reductions
+
+      ``+`` represents ``Nat.add``:
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:220]: Maybe say "is a notation for" (with a link to the notations chapter) instead of "represents". And maybe add a sentence to note that it is defined as a fixpoint (`fix`).

File: doc/sphinx/language/core/conversion.rst [hunk 13]
```diff
@@ -4,11 +4,81 @@ Conversion rules
...
+      :header-rows: 1
+
+      * - Conversion name
+        - Description
+
+      * - beta-reduction
+        - eliminates `fun`
+
+      * - delta-reduction
+        - replaces a defined symbol with its definition
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:55]: I'm not so fond of the use of the undefined word "symbol" in this table. What about writing "defined variable or constant here"?

File: doc/sphinx/language/core/conversion.rst [hunk 14]
```diff
@@ -4,11 +4,81 @@ Conversion rules
...
+
+      * - cofix-reduction
+        - replaces a `cofix` symbol with a :term:`beta-redex`;
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:58]: "co-fixpoint" here

File: doc/sphinx/language/core/conversion.rst [hunk 15]
```diff
@@ -4,11 +4,81 @@ Conversion rules
...
+          recursive calls to the symbol are replaced with the `cofix` term
+
+      * - fix-reduction
+        - replaces a `fix` symbol with a :term:`beta-redex`;
```
Comment (Zimmi48) [inline doc/sphinx/language/core/conversion.rst:62]: and "fixpoint" here.

File: doc/sphinx/proof-engine/ssreflect-proof-language.rst [hunk 16]
```diff
@@ -1202,7 +1202,7 @@ The move tactic.
          Goal not False.
          move.
 
-   More precisely, the :tacn:`move` tactic inspects the goal and does nothing
+   More precisely, the :tacn:`move (ssreflect)` tactic inspects the goal and does nothing
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/ssreflect-proof-language.rst:1205]: What is the relation of this change with the changes to the conversion chapter? I'm OK for this change but please keep it in a separate commit, make sure that all references to this tactic in this chapter are changed, and maybe use `<move (ssreflect)>` to avoid having the parenthesis shown in the text.

File: doc/tools/coqrst/coqdomain.py [hunk 17]
```diff
@@ -1180,8 +1180,7 @@ def ci_sort(entry):
             if type == 'term':
                 entries = content[itemname[0].lower()]
                 entries.append([itemname, 0, docname, anchor, '', '', ''])
-        content = sorted(content.items())
-        return content, False
+        return content.items(), False
```
Comment (Zimmi48) [inline doc/tools/coqrst/coqdomain.py:1183]: What is this about?

File: doc/sphinx/language/core/conversion.rst (hunk 8 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst:43]: I'm fine with "useful".

>I could add a clause saying they can all be applied with conversion tactics, except for eta-expansion.

Yes, that would be good.

File: doc/sphinx/language/core/conversion.rst (hunk 7 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: Looks better, thanks!

File: doc/sphinx/language/core/conversion.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: >There were similar changes in #13707 that I didn't clean up after the rebase.

Do you mean that we should carefully proofread the new state of this chapter?

File: doc/sphinx/language/core/conversion.rst [hunk 18]
```diff
@@ -4,11 +4,83 @@ Conversion rules
 ----------------
 
 Coq has conversion rules that can be used to determine if two
-terms are equal by definition, or :term:`convertible`.
+terms are equal by definition in |CiC|, or :term:`convertible`.
 Conversion rules consist of reduction rules and expansion rules.
+
+Conversion rules can be used
+to determine if two terms are convertible by
```
[the remark this answers is not in the document]
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: Looks good.

**Comment by Zimmi48** [comment]
Note that you have a typo in your commit message.

**Comment by Zimmi48** [comment]
Apart from the comments I made, this looks ready to me.

**Comment by Zimmi48** [comment]
The `needs: squashing` is for the review comments, but keeping unrelated changes in separate commits would still be good.

File: doc/sphinx/language/core/conversion.rst [hunk 19]
```diff
@@ -4,11 +4,81 @@ Conversion rules
...
+      * - cofix-reduction
+        - replaces a `cofix` symbol with a :term:`beta-redex`;
+          recursive calls to the symbol are replaced with the `cofix` term
+
+      * - fix-reduction
+        - replaces a `fix` symbol with a :term:`beta-redex`;
+          recursive calls to the symbol are replaced with the `fix` term
+
+      * - match-reduction
+        - eliminates `match`
```
[the remark this answers is not in the document]
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:66]: You suggest dropping "Reductions that have a unique inverse
expansion are also referred to as *contractions*."?

[the remark this answers is not in the document]
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:66]: I reordered those 3 and moved match..iota to the end of the list (more or less Greek alphabetic order).

File: doc/sphinx/language/core/conversion.rst (hunk 12 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:220]: Think you mean ```:cmd:`Fixpoint` ```

File: doc/sphinx/language/core/conversion.rst (hunk 15 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:62]: replacing "fix-recution" with "fixpoint-reduction"?

File: doc/sphinx/proof-engine/ssreflect-proof-language.rst (hunk 16 above)
Comment (jfehrle) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst:1205]: I used <move (ssreflect)>.  Making it a separate commit (or worse, a separate PR) seems like overkill.  It's akin to fixing a misspelling when you notice it.  Make typos too tedious to fix and they will remain unfixed.  You think?

File: doc/tools/coqrst/coqdomain.py (hunk 17 above)
Comment (jfehrle) [reply doc/tools/coqrst/coqdomain.py:1183]: IIRC it's a superfluous sort.  Removing it makes no difference to the order of glossary entries, which is determined by the "for ... sorted" statement above.

File: doc/sphinx/language/core/conversion.rst (hunk 6 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst:34]: So no change needed here, right?

File: doc/sphinx/language/core/conversion.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst]: I wouldn't bother.  It was just a rebase glitch that I cleaned up.  Hard to remember much 2 months after my comment.

**Comment by jfehrle** [comment]
Updated.  A couple little questions and still needs to be squashed.

File: doc/sphinx/proof-engine/ssreflect-proof-language.rst (hunk 16 above)
Comment (Zimmi48) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst:1205]: The fact that you find it tedious to move it to a separate commit means that you have new git skills to learn. I would agree that it would be overkill and would discourage typo fixing if it was actually difficult. And it would also be a bad thing to make this a requirement for occasional contributors. But since you are a major contributor, I find it worth to tell you that we usually prefer commits that are well separated semantic units (I'm not talking about separate PRs for trivialities here).

In this case, separating the PR in three commits is particularly easy since the changes to separate are in separate files. This is just about doing the right `git add` followed by `git commit`. But even when the changes are in the same file (but in distinct places), it can be relatively easy to do (see https://nuclearsquid.com/writings/git-add/).

Since it's slightly more difficult to do after the fact that during the preparation of the commits, I'll do it for you in this PR.

Comment (Zimmi48) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst]: I've put the main commit last so that it is easier to edit / squash for you later. Remember to pull the new version of the PR.

>I used <move (ssreflect)>.

You didn't do this change, so I did it for you.

File: doc/sphinx/language/core/conversion.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: > You suggest dropping "Reductions that have a unique inverse
> expansion are also referred to as _contractions_."?

No, I don't suggest anything like that.

I guess this comment was in reply to my previous comment "On the contrary...". Note that the reason why you can't comment directly below is because my comment was in reply to a previous thread (to which you also replied).

File: doc/sphinx/language/core/conversion.rst (hunk 6 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst:33]: On the contrary, I was still suggesting inserting `:term:` here. But if you don't like this proposal, I'm not going to fight :wink:

File: doc/sphinx/language/core/conversion.rst (hunk 12 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: In practice, it is defined with the `Fixpoint` command. But what we see when doing the `Print` below is the resulting `fix` operator. That's why I was suggesting to mention that `Nat.add` is defined as a fixpoint (the actual command used doesn't matter). But only do it if you are comfortable writing about this.

File: doc/sphinx/language/core/conversion.rst (hunk 15 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: No, replacing "`fix` symbol". By suggestions here were all about getting rid of the word "symbol".

File: doc/sphinx/language/core/conversion.rst (hunk 12 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/conversion.rst]: OK, actually you've already added something. LGTM.

File: doc/sphinx/proof-engine/ssreflect-proof-language.rst (hunk 16 above)
Comment (jfehrle) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst]: I definitely made that change but somehow lost it.

`git add --patch` is itself tedious, particularly if there are many changes to include or exclude.  And a command-diff is often not very usable.  There were a few other link labels such as `apply (ssreflect)` that I meant to fix.  Grouping those now with the other in the second commit now requires a rebase.  I think these are trivialities.  But we don't have to agree 100%.

Comment (jfehrle) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst]: But squashing at this time was not helpful.  I need to see what I changed in the last round of changes to make sense of your comments.  I refer to the review commits constantly.

File: doc/sphinx/language/core/conversion.rst (hunk 15 above)
Comment (jfehrle) [reply doc/sphinx/language/core/conversion.rst]: Got it.

**Comment by jfehrle** [comment]
Updated.  Fixed fix/cofix and a couple more SSR tactic link labels.  So I think ready to merge.

File: doc/sphinx/proof-engine/ssreflect-proof-language.rst (hunk 16 above)
Comment (Zimmi48) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst]: Ah sorry about that. I thought the review comments were meant for the reviewers only. I didn't realize they were useful for you as well.

**Review by Zimmi48 — APPROVED** [review]
Wonderful!

@coqbot merge now

**Comment by jfehrle** [comment]
Great, thanks to you and @JasonGross for your help!
